### ISSUE TITLE
Proper errors for git merge conflict tokens in code

### ISSUE THREAD

**Issue by SnirkImmington** [immington-industries/protosnirk#64]
The protosnirk parser recognizes the symbol `<<<<<<<` as `GitMergeBegin`. This is inspired by [this LLVM blog post](http://blog.llvm.org/2010/04/amazing-feats-of-clang-error-recovery.html) about a clang feature which gives a better error message for a merge conflict indicator in a source file:
```
$ clang t.c
t.c:2:1: error: version control conflict marker in file
<<<<<<< HEAD
^

$ gcc t.c
t.c: In function ‘f0’:
t.c:2: error: expected expression before ‘<<’ token
t.c:4: error: expected expression before ‘==’ token
t.c:6: error: expected expression before ‘>>’ token
```
(The blog post is 8 years old right now, for all I know GCC has since added this feature.)

The protosnirk compiler will at the very least give a "we found `<<<<<<<`" looking error, but we don't do anything else with it. 

Once we have better error messages, we should include this special token in error message handling.

**Comment by Timidger** [comment]
FYI that particular syntax is a property of the `diff3` utility program, not from git itself. Mercurial, for example, uses the same format for three way merge conflicts.

**Comment by SnirkImmington** [comment]
Yeah, the name in the source code isn't good - this is why Clang's error is "version control conflict marker."

**Comment by Timidger** [comment]
Yeah, just thought I'd point it out so you don't emit an error talking about git and confuse all 7 of the Mercurial users in existence ;)
